### PR TITLE
[Go]: Resolve conflict when a multi-keyed list contains a container named "key"

### PR DESCRIPTION
**Pull request by wenovus**
Do this by keeping the current naming, but change to "%s_%s_key" instead of "%s_%s_Key" as the back up name if a conflict like this occurs. This should always work since child non-leaves are named using camel cases, so I'm making it error out if the backup is already taken.

Tested:
- Added a new integration test for this case.

### CODE REVIEW

File: ygen/gogen.go [hunk 1]
```diff
@@ -2008,6 +2008,13 @@ func yangListFieldToGoType(listField *yang.Entry, listFieldName string, parent *
 		// generatedGoMultiKeyListStruct struct, which is then expanded by a template to the struct
 		// definition.
 		listKeyStructName = fmt.Sprintf("%s_%s_Key", parent.Name, listFieldName)
+		if gogen.definedGlobals[listKeyStructName] {
+			listKeyStructName = fmt.Sprintf("%s_%s_key", parent.Name, listFieldName)
```
Comment (robshakir) [inline ygen/gogen.go:2012]: I think this is non-obvious to the user, the name difference `Foo_Key` vs `Foo_key` in a long name seems insufficient, as well as not auto-complete friendly.

I would suggest that maybe use something like `YANGListKey` at this stage -- we tried to do the nice thing of having it just be `Key`, but given that the schema wouldn't let us do that then going with a longer-but-more-explicit name might be best?

**Review by robshakir — COMMENTED** [review]
The change looks good -- just one discussion for us to have about how to name things in this case.

File: ygen/gogen.go (hunk 1 above)
Comment (wenovus) [reply ygen/gogen.go:2012]: Makes sense, the user might be confused what the difference between "_key" and "_Key" are, but a more explicit name makes that clear.

Comment (robshakir) [reply ygen/gogen.go]: Great, thanks -- can we update [the docs](https://github.com/openconfig/ygot/blob/master/docs/design.md#yang-lists) to specify that this is the case, just so there's some reference for users?

Comment (wenovus) [reply ygen/gogen.go]: Done.

**Review by robshakir — APPROVED** [review]
Thanks for the changes!

**Pull request merged**